File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/Latch.java
package org.firstinspires.ftc.teamcode.utilities;

import static com.qualcomm.robotcore.hardware.DcMotor.ZeroPowerBehavior.BRAKE;

import com.acmerobotics.dashboard.config.Config;
import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;
import com.qualcomm.robotcore.hardware.DistanceSensor;

//:)

import org.firstinspires.ftc.robotcore.external.Telemetry;
import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit;
    @Config
    public class Latch {
        Servo latchServo;
        public Latch(HardwareMap hmap, Telemetry telemetry) {
            this.latchServo = hmap.servo.get(CONFIG.latch);
            this.telemetry = telemetry;
        }
        public static double latched = 1.0;
        public static double unlatched = 0.8;

        public static Telemetry telemetry;

        public void latchOn() {
            telemetry.addLine("LatchOn, Dpad Left Button");
            latchServo.setPosition(latched);
        }
        public void latchOff() {
            telemetry.addLine("Latch off, Dpad Right button");
            latchServo.setPosition(unlatched);
        }
        public double getPosition() {
            return (latchServo.getPosition());
        }

    }

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/Latch.kt
package org.firstinspires.ftc.teamcode.utilities

import com.acmerobotics.dashboard.config.Config
import com.qualcomm.robotcore.hardware.HardwareMap
import com.qualcomm.robotcore.hardware.Servo
import org.firstinspires.ftc.robotcore.external.Telemetry

//:)
@Config
class Latch(hmap: HardwareMap, var telemetry: Telemetry) {
    var latchServo: Servo = hmap.servo[CONFIG.latch]
    fun latchOn() {
        Companion.telemetry.addLine("latch on")
        Companion.telemetry.addLine(latchServo.position.toString())
        latchServo.position = latched
    }

    fun latchOff() {
        Companion.telemetry.addLine("latch off")
        Companion.telemetry.addLine(latchServo.position.toString())
        latchServo.position = unlatched
    }

    val position: Double
        get() = (latchServo.position)

    companion object {
        var latched: Double = 1.0 // -0.70
        var unlatched: Double = 0.6
    }
}
